xrpl/utils/tx_parser/balance_changes.py

```diff
@@ -0,0 +1,332 @@
+                            str,
+                            Union[
+                                str,
+                                int,
+                            ],
+                        ],
+                    ],
+                ],
+                Dict[  # Field: 'meta'
+                    str,  # 'AffectedNodes'
+                    List[
+                        Dict[
+                            str,  # Node state
+                            Dict[
+                                str,
+                                Union[
+                                    str,
+                                    int,
+                                    Dict[str, Union[str, int, Dict[str, str]]],
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ],
+    ) -> None:
+        """
+        Attributes:
+            metadata:
+                Transaction metadata including the account that
+                sent the transaction and the affected nodes.
+
+        Usage:
+            Parse all final balances:
+                `ParseFinalBalances(metadata=tx).all_balances`
+            Parse final balances of the account that sent the transaction:
+                `ParseFinalBalances(metadata=tx).source_account_balances`
+            Parse final balances of the account that received the transaction
+                `ParseFinalBalances(metadata=tx).destination_account_balances``
+                    If provided transaction has no field `Destination`:
+                        `None`
+        """
+        super().__init__(metadata)
+
+        self._parse()
+
+    def _parse(self: ParseFinalBalances) -> None:
+        is_valid_metadata(metadata=self.metadata)
+        if "transaction" in self.metadata:
+            self.metadata = normalize_metadata(self.metadata)
+
+        nodes = normalize_nodes(metadata=self.metadata)
+        parsed_quantities = parse_quantities(
+            nodes=nodes,
+            value_parser=parse_final_balance,
+        )
+
+        result: Dict[str, List[Dict[str, str]]] = {}
+        for address, change in parsed_quantities.items():
+            result[address] = []
+            for obj in change:
+                if isinstance(obj.counterparty, tuple):
+                    obj.counterparty = obj.counterparty[0]
+                result[address].append(
+                    {
+                        "Counterparty": obj.counterparty,
+                        "Currency": obj.currency,
+                        "Value": obj.value,
+                    }
+                )
+
+        self._define_results(result=result)
+
+
+class ParsePreviousBalances(BalanceChanges):
+    """Parse the previous balances of all accounts affected
+    by the transaction before it occurred.
+    """
+
+    def __init__(
+        self: ParsePreviousBalances,
+        metadata: Dict[  # metadata received from a tx method
```

Comment:
You can define this type at the top or in your utils file, which will make it much neater.

Reply:
e.g.
```py
Metadata = Dict[str, ...]
```

Reply:
You can even nest types to make it neater